<launch>
    <arg name="robot"      default="dk1_right" doc="The robot to run. Can be _left/_right of: dk1, nova, unknown."/>
    <arg name="gloveIndex" default="1"         doc="specify the number of the glove, it's an identifier. Not the amount of gloves connected."/>
    <arg name="isRight"    default="true"      doc="specify the righthandedness of the glove"/>

    <group ns="$(eval '/senseglove/' + str(int(arg('gloveIndex')/2)) + '/rh')" if="$(eval isRight)">

        <param name="robot_description" textfile="$(find senseglove_description)/urdf/$(arg robot).urdf" if="$(eval arg('isRight')==true)"/>
        <rosparam file="$(find senseglove_hardware_interface)/config/$(arg robot)/controllers.yaml" command="load"/>

        <node
            name="robot_state_publisher" 
            pkg="robot_state_publisher"
            type="robot_state_publisher">
            <param name="publish_frequency" value="30"/>
        </node>

        <node 
            name="controller_spawner" 
            pkg="controller_manager"
            type="controller_manager"
            ns="$(eval '/senseglove/' + str(int(arg('gloveIndex')/2)) + '/rh')"
            args="spawn controller/joint_state controller/trajectory"
            output="screen"
            respawn="false">
        </node>

        <node
            name="hardware_interface"
            pkg="senseglove_hardware_interface"
            type="senseglove_hardware_interface_node"
            args="$(arg robot) $(arg gloveIndex) $(arg isRight)"
            output="screen"
            required="true">
        </node>
    </group>

    <group ns="$(eval '/senseglove/' + str(int(arg('gloveIndex')/2)) + '/lh')" unless="$(eval isRight)">

        <param name="robot_description" textfile="$(find senseglove_description)/urdf/$(arg robot).urdf" if="$(eval arg('isRight')==false)"/>
        <rosparam file="$(find senseglove_hardware_interface)/config/$(arg robot)/controllers.yaml" command="load"/>

        <node
            name="robot_state_publisher"
            pkg="robot_state_publisher"
            type="robot_state_publisher">
            <param name="publish_frequency" value="30"/>
        </node>

        <node
            name="controller_spawner"
            pkg="controller_manager"
            type="controller_manager"
            ns="$(eval '/senseglove/' + str(int(arg('gloveIndex')/2)) + '/lh')"
            args="spawn controller/joint_state controller/trajectory"
            output="screen"
            respawn="false">
        </node>

        <node
            name="hardware_interface"
            pkg="senseglove_hardware_interface"
            type="senseglove_hardware_interface_node"
            args="$(arg robot) $(arg gloveIndex) $(arg isRight)"
            output="screen"
            required="true">
        </node>
    </group>    
</launch>
